components/activity_log_modal/activity_log_modal.jsx

```diff
@@ -136,189 +138,81 @@ export default class ActivityLogModal extends React.Component {
     };
 
     render() {
-        const activityList = [];
-
-        for (let i = 0; i < this.state.sessions.length; i++) {
-            const currentSession = this.state.sessions[i];
-            const lastAccessTime = new Date(currentSession.last_activity_at);
-            const firstAccessTime = new Date(currentSession.create_at);
-            let devicePlatform = currentSession.props.platform;
-            let devicePicture = '';
-            let deviceTitle = '';
-
-            if (currentSession.props.type === 'UserAccessToken') {
-                continue;
-            }
+        let content;
+        if (this.props.sessions.loading) {
+            content = <LoadingScreen/>;
+        } else {
+            const activityList = this.props.sessions.reduce((array, currentSession, index) => {
+                const lastAccessTime = new Date(currentSession.last_activity_at);
+                const firstAccessTime = new Date(currentSession.create_at);
```

Comment:
If `new Date` is unavoidable inside the render function, then I'm thinking that we should only do it as close as the leaf component. So in this case, conversion of `lastAccessTime` in `ActivityLog` component while `firstAccessTime` inside `MoreInfo` component.